eng/scripts/Set-TemplateTestMatrixVariable.ps1

```diff
@@ -72,6 +72,12 @@ if ($TemplateListFilter -ne '.*') {
     $templateNames = $templateNames -match $TemplateListFilter
 }
 
+# If we only had a single template name, templateNames may be a string at this point
+# but we really want it to be an array, for the rest of the script.
+if ($templateNames -is [string]) {
```

Comment:
@danieljurek @weikanglim 

Was trying to fix this issue: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1990801&view=logs&j=3343d4d0-7547-5e7f-3a13-3ce10a15fbf0&t=fb69b4da-561b-5f99-1833-bbac10238ce3&l=13

But my powershell foo is not great here.  It seems like `.Split()` can return a string if there's just a single item, but we really expect it to be an array.

If there's a better way to do this - I'll all ears.

Reply:
@ellismg Thanks for fixing my sloppy code.

With the current approach, we're still too late if `TemplateList` had one entry:
`Set-TemplateTestMatrixVariable.ps1 -TemplateListFilter "csharp-cosmos-sql" -TemplateList "Azure-Samples/todo-csharp-cosmos-sql"` fails because we would to a `string` -match `string` which returns `true`/`false`

I think the more robust way is to make `$templateNames` an array by default, and to use the append operator `+=` to force string coercion to array if necessary. This way we guarantee the variable is always an `array` throughout.

```pwsh
$templateNames = @()

if ($TemplateList -eq '(azd template list)') {
    Write-Host "Using results of (azd template list --output json)"
    
    $templateNames += (azd template list --output json | ConvertFrom-Json).name
    if ($LASTEXITCODE -ne 0) {
        Write-Error "azd template list failed"
        exit 1
    }
} else {
    Write-Host "Using provided TemplateList value: $TemplateList"

    $templateNames += ($TemplateList -split ",").Trim()
}
```

Reply:
Thanks, @weikanglim - I adopted these changes.